programs/relayer/src/modules/cosmos_to_cosmos.rs

```diff
@@ -0,0 +1,194 @@
+    },
+    core::modules::ModuleServer,
+};
+
+/// The `CosmosToCosmosRelayerModule` struct defines the Cosmos to Cosmos relayer module.
+#[derive(Clone, Copy, Debug)]
+#[allow(clippy::module_name_repetitions)]
+pub struct CosmosToCosmosRelayerModule;
+
+/// The `CosmosToCosmosRelayerModuleServer` defines the relayer server from Cosmos to Cosmos.
+#[allow(dead_code)]
+struct CosmosToCosmosRelayerModuleServer {
+    /// The souce chain listener for Cosmos SDK.
+    pub src_listener: cosmos_sdk::ChainListener,
+    /// The target chain listener for Cosmos SDK.
+    pub target_listener: cosmos_sdk::ChainListener,
+    /// The transaction builder from Cosmos to Cosmos.
+    pub tx_builder: cosmos_to_cosmos::TxBuilder,
+}
+
+/// The configuration for the Cosmos to Cosmos relayer module.
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize)]
+#[allow(clippy::module_name_repetitions)]
+pub struct CosmosToCosmosConfig {
+    /// The source tendermint RPC URL.
+    pub src_rpc_url: String,
+    /// The target tendermint RPC URL.
+    pub target_rpc_url: String,
+    /// The address of the submitter.
+    /// Required since cosmos messages require a signer address.
+    pub signer_address: String,
+}
+
+impl CosmosToCosmosRelayerModuleServer {
+    #[allow(dead_code)]
+    fn new(config: CosmosToCosmosConfig) -> Self {
+        let src_client = HttpClient::new(
+            Url::from_str(&config.src_rpc_url)
+                .unwrap_or_else(|_| panic!("invalid tendermint RPC URL: {}", config.src_rpc_url)),
+        )
+        .expect("Failed to create tendermint HTTP client");
+
+        let src_listener = cosmos_sdk::ChainListener::new(src_client.clone());
+
+        let target_client =
+            HttpClient::new(Url::from_str(&config.target_rpc_url).unwrap_or_else(|_| {
+                panic!("invalid tendermint RPC URL: {}", config.target_rpc_url)
+            }))
+            .expect("Failed to create tendermint HTTP client");
+
+        let target_listener = cosmos_sdk::ChainListener::new(target_client.clone());
+
+        let tx_builder =
+            cosmos_to_cosmos::TxBuilder::new(src_client, target_client, config.signer_address);
+
+        Self {
+            src_listener,
+            target_listener,
+            tx_builder,
+        }
+    }
+}
+
+#[tonic::async_trait]
+impl RelayerService for CosmosToCosmosRelayerModuleServer {
+    #[tracing::instrument(skip_all)]
+    async fn info(
+        &self,
+        _request: Request<api::InfoRequest>,
+    ) -> Result<Response<api::InfoResponse>, tonic::Status> {
+        tracing::info!("Received info request.");
+        Ok(Response::new(api::InfoResponse {
+            target_chain: Some(api::Chain {
+                chain_id: self
+                    .target_listener
+                    .chain_id()
+                    .await
+                    .map_err(|e| tonic::Status::from_error(e.to_string().into()))?,
+                ibc_version: "2".to_string(),
```

Comment:
I guess we only support version 2, do we have any checks anywhere to fail during setup if version 1 comes in?